Update copyright statements

franzliedke opened the pull request:
This updates the copyright statements, as the project has now been transferred to the Flarum Foundation (Stichting Flarum).

This is my proposal. Let's discuss the header we want here - code changes can then be enforced with StyleCI.

This needs approval from @tobyzerner, too.

franzliedke commented:
Haha, I knew this would happen. :roll_eyes: 

Please review the commits one-by-one - StyleCI already reverted the change to the code file, so the total diff does not show my complete proposal.

tobyzerner commented:
Looks good to me 👍

franzliedke commented:
Open questions from my side:
- "Stichting Flarum" or "Flarum Foundation"?
- Does the "based on code" make sense in the `LICENSE` file?

jordanjay29 commented:
> Open questions from my side:
> 
>     * "Stichting Flarum" or "Flarum Foundation"?

I think we definitely need the former for legal clarity, but the latter is more clear for most non-Dutch people. Perhaps `Stichting Flarum (Flarum Foundation)`

>     * Does the "based on code" make sense in the `LICENSE` file?

No, if I didn't know the context, I would have no idea what this meant. It might be better to append the years of those copyrights such as: 
```
Copyright (c) 2019 Stichting Flarum 
Copyright (c) 2014-2019 Toby Zerner (toby.zerner@gmail.com)
```

datitisev commented:
Not sure if this matters, but it'd probably make more sense to not include the "based on code" in the files themselves. The license comment does reference the LICENSE file, so not sure if that's necessary or if we just want it there.

Also yeah, I agree that copyright years might be better.

luceos commented:
I'm with @jordanjay29 on this.

franzliedke commented:
Okay, I added the years to the LICENSE file, and removed copyright information from the file headers. @tobyzerner mentioned this possibility in chat at some time.

Is that fine for everyone?

jordanjay29 commented on LICENSE:
```diff
@@ -1,6 +1,7 @@
 The MIT License (MIT)
 
-Copyright (c) Toby Zerner
+Copyright (c) 2019 Stichting Flarum
```
```suggestion
Copyright (c) 2019 Stichting Flarum (Flarum Foundation)
```

tobyzerner commented:
👍

franzliedke commented:
Okay, I configured the proposed code file header in StyleCI, but it didn't apply it. :thinking:

franzliedke commented:
Ah, apparently StyleCI needed a commit to trigger the change. So I pushed [the LICENSE file change](https://github.com/flarum/core/commit/19188e3eda6b34106c6f771e4ff15f219bc57270) to master now, and it followed up with [the code file changes](https://github.com/flarum/core/commit/d492579638fb52dafbfe65f1f36a95eb6047f7f3).
